lib/smart_name/manipulate.rb

```diff
@@ -0,0 +1,70 @@
+class SmartName
+  module Manipulate
+    # replace a subname
+    # keys are used for comparison
+    def replace old, new
+      old_name = old.to_name
+      new_name = new.to_name
+      return self if old_name.length > length
+      return replace_part(old_name, new_name) if old_name.simple?
+      return self unless include? old_name
+      replace_all_subsequences(old_name.parts, new_name.parts).to_name
+    end
+
+    def replace_part oldpart, newpart
+      ensure_simpliness oldpart, "Use 'replace' to replace junctions"
```

Comment:
`replace` acts as the old `replace_part` and `replace_part` now replaces only parts